### ISSUE TITLE
OCF often stuck when using cores over 16

### ISSUE THREAD

**Issue by USTCqzy** [spdk/spdk#2134]
# Bug report

OCF often stuck when core num over 16 (not 100%).
It is hard for me to find out the reason, for it is just **stuck forever**.

## Expected Behavior

OCF works.

## Current Behavior

OCF often stuck when core num over 16 (not 100%).
PS: `stuck` meams it seems like running, but never end, so you can imagine there are some messages print successfully before `Running I/O for xxx seconds...`
For example:
```bash
[2021-08-27 06:02:00.586057] Starting SPDK v21.10-pre git sha1 1763a126c / DPDK 21.05.0 initialization...
[2021-08-27 06:02:00.586149] [ DPDK EAL parameters: [2021-08-27 06:02:00.586169] bdevperf [2021-08-27 06:02:00.586185] --no-shconf [2021-08-27 06:02:00.586199] -c 0xFFFF [2021-08-27 06:02:00.586214] --log-level=lib.eal:6 [2021-08-27 06:02:00.586229] --log-level=lib.cryptodev:5 [2021-08-27 06:02:00.586243] --log-level=user1:6 [2021-08-27 06:02:00.586259] --iova-mode=pa [2021-08-27 06:02:00.586274] --base-virtaddr=0x200000000000 [2021-08-27 06:02:00.586289] --match-allocations [2021-08-27 06:02:00.586304] --file-prefix=spdk_pid1388650 [2021-08-27 06:02:00.586319] ]
EAL: No available 1048576 kB hugepages reported
TELEMETRY: No legacy callbacks, legacy socket not created
[2021-08-27 06:02:00.690045] app.c: 540:spdk_app_start: *NOTICE*: Total cores available: 16
[2021-08-27 06:02:00.842093] reactor.c: 936:reactor_run: *NOTICE*: Reactor started on core 1
[2021-08-27 06:02:00.842177] reactor.c: 936:reactor_run: *NOTICE*: Reactor started on core 2
[2021-08-27 06:02:00.842255] reactor.c: 936:reactor_run: *NOTICE*: Reactor started on core 3
[2021-08-27 06:02:00.842350] reactor.c: 936:reactor_run: *NOTICE*: Reactor started on core 4
[2021-08-27 06:02:00.842427] reactor.c: 936:reactor_run: *NOTICE*: Reactor started on core 5
[2021-08-27 06:02:00.842506] reactor.c: 936:reactor_run: *NOTICE*: Reactor started on core 6
[2021-08-27 06:02:00.842586] reactor.c: 936:reactor_run: *NOTICE*: Reactor started on core 7
[2021-08-27 06:02:00.842646] reactor.c: 936:reactor_run: *NOTICE*: Reactor started on core 8
[2021-08-27 06:02:00.842678] reactor.c: 936:reactor_run: *NOTICE*: Reactor started on core 9
[2021-08-27 06:02:00.842756] reactor.c: 936:reactor_run: *NOTICE*: Reactor started on core 10
[2021-08-27 06:02:00.842836] reactor.c: 936:reactor_run: *NOTICE*: Reactor started on core 11
[2021-08-27 06:02:00.842942] reactor.c: 936:reactor_run: *NOTICE*: Reactor started on core 13
[2021-08-27 06:02:00.843019] reactor.c: 936:reactor_run: *NOTICE*: Reactor started on core 14
[2021-08-27 06:02:00.842915] reactor.c: 936:reactor_run: *NOTICE*: Reactor started on core 12
[2021-08-27 06:02:00.843103] reactor.c: 936:reactor_run: *NOTICE*: Reactor started on core 0
[2021-08-27 06:02:00.843102] reactor.c: 936:reactor_run: *NOTICE*: Reactor started on core 15
[2021-08-27 06:02:00.843620] accel_engine.c: 988:spdk_accel_engine_initialize: *NOTICE*: Accel engine initialized to use software engine.
[2021-08-27 06:02:01.020157] 'OCF Core' volume operations registered
[2021-08-27 06:02:01.023194] 'SPDK block device' volume operations registered
[2021-08-27 06:02:04.262649] nvme_ctrlr.c:2742:nvme_ctrlr_construct_namespaces: *NOTICE*: nn 1, mem size 8256
[2021-08-27 06:02:04.281748] Inserting cache Cache1
[2021-08-27 06:02:04.282112] Cache1: Metadata initialized
[2021-08-27 06:02:04.282199] Cache1: Successfully added
[2021-08-27 06:02:04.282204] Cache1: Cache mode : wt
[2021-08-27 06:02:04.287619] Cache1: Super block config offset : 0 kiB
[2021-08-27 06:02:04.287626] Cache1: Super block config size : 2196 B
[2021-08-27 06:02:04.287628] Cache1: Super block runtime offset : 128 kiB
[2021-08-27 06:02:04.287630] Cache1: Super block runtime size : 4 B
[2021-08-27 06:02:04.287632] Cache1: Reserved offset : 256 kiB
[2021-08-27 06:02:04.287634] Cache1: Reserved size : 128 kiB
[2021-08-27 06:02:04.287636] Cache1: Part config offset : 384 kiB
[2021-08-27 06:02:04.287639] Cache1: Part config size : 48 kiB
[2021-08-27 06:02:04.287641] Cache1: Part runtime offset : 512 kiB
[2021-08-27 06:02:04.287643] Cache1: Part runtime size : 28 kiB
[2021-08-27 06:02:04.287645] Cache1: Core config offset : 640 kiB
[2021-08-27 06:02:04.287647] Cache1: Core config size : 484 kiB
[2021-08-27 06:02:04.287649] Cache1: Core runtime offset : 1152 kiB
[2021-08-27 06:02:04.287651] Cache1: Core runtime size : 1172 kiB
[2021-08-27 06:02:04.287653] Cache1: Core UUID offset : 2432 kiB
[2021-08-27 06:02:04.287655] Cache1: Core UUID size : 16384 kiB
[2021-08-27 06:02:04.287657] Cache1: Cleaning offset : 18816 kiB
[2021-08-27 06:02:04.287659] Cache1: Cleaning size : 2876 kiB
[2021-08-27 06:02:04.287661] Cache1: Eviction offset : 21760 kiB
[2021-08-27 06:02:04.287663] Cache1: Eviction size : 1916 kiB
[2021-08-27 06:02:04.287665] Cache1: Collision offset : 23680 kiB
[2021-08-27 06:02:04.287667] Cache1: Collision size : 10112 kiB
[2021-08-27 06:02:04.287669] Cache1: List info offset : 33792 kiB
[2021-08-27 06:02:04.287672] Cache1: List info size : 4088 kiB
[2021-08-27 06:02:04.287674] Cache1: Hash offset : 37888 kiB
[2021-08-27 06:02:04.287676] Cache1: Hash size : 240 kiB
[2021-08-27 06:02:04.287678] Cache1: Cache line size: 64 kiB
[2021-08-27 06:02:04.287680] Cache1: Metadata capacity: 37 MiB
[2021-08-27 06:02:04.291053] Cache1: OCF metadata self-test PASSED
[2021-08-27 06:02:04.378046] Cache1: Policy 'always' initialized successfully
[2021-08-27 06:02:04.483633] Cache1: Done saving cache state!
[2021-08-27 06:02:04.492214] Cache1: Cache attached
[2021-08-27 06:02:04.492311] Cache1: Successfully attached
[2021-08-27 06:02:04.492511] Cache1: Inserting core NVMe1n1
[2021-08-27 06:02:04.492520] Cache1.NVMe1n1: Seqential cutoff init
[2021-08-27 06:02:04.500674] Cache1.NVMe1n1: Successfully added
```

Let me introduce some background:
- DRAM 15GB+Optane SSD
- io_depth=1
- threads=("0x1" "0xFF"  "0xFFFF" "0xFFFFFFFF" "0xFFFFFFFFFFFFFFFF")
- io_size=512, 4k, 16k, 64k
- cache_line_size=4k, 8k, 16k, 32k, 64k
- cache_mode=wt
- config example:

test.json
```json
{
    "subsystems": [
      {
        "subsystem": "bdev",
        "config": [
          {
            "method": "bdev_malloc_create",
            "params": {
              "name": "Malloc0",
              "num_blocks": 31457280,
              "block_size": 512,
              "uuid": "a2e6d1bb-1cf5-484a-bd4a-ebbff0043c8b"
            }
          },
          {
            "method": "bdev_nvme_attach_controller",
            "params": {
              "name": "NVMe1",
              "trtype": "PCIe",
              "traddr": "0000:9b:00.0",
              "prchk_reftag": false,
              "prchk_guard": false
            }
          },
          {
            "method": "bdev_ocf_create",
            "params": {
              "name": "Cache1",
              "mode": "wt",
              "cache_line_size": 64,
              "cache_bdev_name": "Malloc0",
              "core_bdev_name": "NVMe1n1"
            }
          }
        ]
      }
    ]
  }
```

I used `timeout -k` to run script, and use `echo Ret is $?` to print error.

As the background showed, I runned 100 tests, and **found 6 of them were killed**, whose configs showed as follow:
```
io depth is 1, read type is randread, thread is 0xFFFF, io size is 65536, cache mode is wt, cache line size is 64
Running I/O for 300 seconds...
Ret is 137
io depth is 1, read type is randread, thread is 0xFFFFFFFF, io size is 512, cache mode is wt, cache line size is 4
Running I/O for 300 seconds...
Ret is 137
io depth is 1, read type is randread, thread is 0xFFFFFFFF, io size is 65536, cache mode is wt, cache line size is 16
Running I/O for 300 seconds...
Ret is 137
io depth is 1, read type is randread, thread is 0xFFFFFFFF, io size is 65536, cache mode is wt, cache line size is 64
Running I/O for 300 seconds...
Ret is 137
io depth is 1, read type is randread, thread is 0xFFFFFFFFFFFFFFFF, io size is 4096, cache mode is wt, cache line size is 8
Running I/O for 300 seconds...
Ret is 137
io depth is 1, read type is randread, thread is 0xFFFFFFFFFFFFFFFF, io size is 16384, cache mode is wt, cache line size is 32
Running I/O for 300 seconds...
Ret is 137
```
In conclusion, it is too complex and really hard for me to point out specific problem (Btw, I am sure that **when # of threads <16, never stuck**, considering of the print messages). _I have tested hundreds of times_, so I am sure there is some problems, which hurt my working efficency as well.

## Possible Solution

No idea.

## Steps to Reproduce

If there is any need, please let me know. (I have a script can be used to reproduce 100 tests showed before. )

## Context (Environment including OS version, SPDK version, etc.)

SPDK master
Ubuntu 20.04

**Comment by gaowayne** [comment]
@USTCqzy could you please let us know if you can close this issue or you have new question on this?

**Comment by USTCqzy** [comment]
@gaowayne Sorry for not having time to test this problem these days. Please close this issue.
